Add sampleAnnot argument, to add sample annotations when reading data

csoneson opened the pull request:
Also add a `trim` argument to `readModBam`, to only return positions inside the specified regions, and a couple of small fixes in other places.

csoneson committed:
Add test for NA names in grl

csoneson committed:
Add sampleAnnot argument to provide sample annotations to read* functions

csoneson committed:
Use all_of to address tidyselect deprecation message

csoneson committed:
Add trim argument to readModBam

csoneson committed:
Use assay colnames in .preparePlotdataReads to deal with situation where some samples were pruned

csoneson committed:
Update NEWS

csoneson committed:
Merge branch 'devel' into add-sample-annot

csoneson committed:
Enable plotting of footprints

csoneson committed:
Merge branch 'devel' into add-sample-annot

csoneson committed:
Don't record the same read-level colData column name multiple times

csoneson committed:
Update NEWS

csoneson committed:
Harmonize man page titles

csoneson committed:
No need to check for 'signal' twice (checked in .assertPackagesAvailable())

csoneson committed:
Remove signal from NAMESPACE

mbstadler reviewed (approved):
Wow - impressive! Looks all great. I may have caused a conflict in `test-plotRegion.R` when I added a `GRangesList` with an `NA` name - sorry about that!

As far as I am concerned feel free to merge and update our installation!

csoneson committed:
Fix merge conflicts

Merge branch 'devel' into add-sample-annot

# Conflicts:
#	R/footprintScoring.R
#	R/plotRegion.R
#	tests/testthat/test-plotRegion.R

csoneson commented:
Merged and installed 🙂
